### PR TITLE
[Merged by Bors] - Add `insert_children` and `push_children` to EntityMut

### PR DESCRIPTION
**Pull request by Davier**
The only API to add a parent/child relationship between existing entities is through commands, there is no easy way to do it from `World`. Manually inserting the components is not completely possible since `PreviousParent` has no public constructor.

This PR adds two methods to set entities as children of an `EntityMut`: `insert_children` and `push_children`. ~~The API is similar to the one on `Commands`, except that the parent is the `EntityMut`.~~ The API is the same as in #1703.
However, the `Parent` and `Children` components are defined in `bevy_transform` which depends on `bevy_ecs`, while `EntityMut` is defined in `bevy_ecs`, so the methods are added to the `BuildWorldChildren` trait instead.
If #1545 is merged this should be fixed too.

I'm aware cart was experimenting with entity hierarchies, but unless it's a coming soon this PR would be useful to have meanwhile.

### CODE REVIEW

File: crates/bevy_transform/src/hierarchy/child_builder.rs [hunk 1]
```diff
@@ -282,6 +284,43 @@ impl<'w> BuildWorldChildren for EntityMut<'w> {
...
+        let world = unsafe { self.world_mut() };
+        for child in children.iter() {
+            world
+                .entity_mut(*child)
+                .insert_bundle((Parent(parent), PreviousParent(parent))); // FIXME: dom't erase the previous parent (see #1545)
+        }
+
+        if let Some(mut children_component) = world.get_mut::<Children>(parent) {
+            children_component.0.insert_from_slice(index, children);
+        } else {
```
Comment (TheRawMeatball) [inline crates/bevy_transform/src/hierarchy/child_builder.rs:318]: Here, we should panic if index != 0

**Review by TheRawMeatball — APPROVED** [review]

File: crates/bevy_transform/src/hierarchy/child_builder.rs [hunk 2]
```diff
@@ -282,6 +284,43 @@ impl<'w> BuildWorldChildren for EntityMut<'w> {
...
+        let world = unsafe { self.world_mut() };
+        for child in children.iter() {
+            world
+                .entity_mut(*child)
+                .insert_bundle((Parent(parent), PreviousParent(parent))); // FIXME: dom't erase the previous parent (see #1545)
+        }
+        if let Some(mut children_component) = world.get_mut::<Children>(parent) {
+            children_component.0.extend(children.iter().cloned());
+        } else {
+            world.entity_mut(parent).insert(Children::with(children));
```
Comment (cart) [inline crates/bevy_transform/src/hierarchy/child_builder.rs:300]: We can optimize this by using self (which is `world.entity_mut(parent)`). Additionally, we haven't updated the entity location, so theres no need to call self.update_location() here.

I'll just push the changes so we can get this merged asap.

Comment (cart) [reply crates/bevy_transform/src/hierarchy/child_builder.rs:300]: I guess I should say, "we haven't modified the entity location using the unsafe world reference". Using self.insert will handle the location stuff for us.

File: crates/bevy_transform/src/hierarchy/child_builder.rs [hunk 3]
```diff
@@ -300,6 +339,48 @@ impl<'w> BuildWorldChildren for WorldChildBuilder<'w> {
         self.current_entity = self.parent_entities.pop();
         self
     }
+
+    fn push_children(&mut self, children: &[Entity]) -> &mut Self {
+        let parent = self
+            .current_entity
+            .expect("Cannot add children without a parent. Try creating an entity first.");
+        self.parent_entities.push(parent);
```
Comment (cart) [inline crates/bevy_transform/src/hierarchy/child_builder.rs:347]: We aren't creating a "nested WorldChildBuilder context" here like we do in `with_children`, so theres no need to push the parent entity (and in fact, that will break things).

**Pull request closed without merging**